### PR TITLE
6497 test_script_save utils to raise potential errors

### LINKED ISSUE #6497: `tests.utils.test_script_save` does not raise error
https://github.com/Project-MONAI/MONAI/blob/6d47cbcfa02d3cbd6f727c3b7abca46c598b6953/tests/utils.py#L743

The reason why I found this bug is that I met torchscript issue in: https://github.com/Project-MONAI/model-zoo/actions/runs/4925952253/jobs/8800809158#step:2:670
but when change the dynunet, the related torchscript test passed.

### PR DESCRIPTION
**Pull request by wyli**
Fixes #6497

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by yiheng-wang-nv** [comment]
Thanks @wyli . May also need to revert the changes in:
https://github.com/Project-MONAI/MONAI/pull/6444
As it broke the torchscript export on DynUNet (did not raise the error because of the bug reported in #6497 )

File: tests/test_ssim_loss.py [hunk 1]
```diff
@@ -47,10 +48,10 @@ def test_shape(self):
         expected_val = [[0.9121], [0.9971]]
         np.testing.assert_allclose(result.detach().cpu().numpy(), expected_val, rtol=1e-4)
 
-    def test_script(self):
-        loss = SSIMLoss(spatial_dims=2)
-        test_input = torch.ones(2, 2, 16, 16)
-        test_script_save(loss, test_input, test_input)
+    # def test_script(self):
+    #     loss = SSIMLoss(spatial_dims=2)
```
Comment (wyli) [inline tests/test_ssim_loss.py:52]: won't be easy to have torchscript support for the current `SSIMLoss`:
```
Module 'SSIMLoss' has no attribute 'ssim_metric' (This attribute exists on the Python module, but we failed to convert Python type: 'monai.metrics.regression.SSIMMetric' to a TorchScript type. Only tensors and (possibly nested) tuples of tensors, lists, or dictsare supported as inputs or outputs of traced functions, but instead got value of type SSIMMetric.. Its type was inferred; try adding a type annotation for the attribute.):
  File "/Documents/MONAI/monai/losses/ssim_loss.py", line 117
                print(1-SSIMLoss(spatial_dims=3)(x,y))
        """
        ssim_value = self.ssim_metric._compute_tensor(input, target).view(-1, 1)
                     ~~~~~~~~~~~~~~~~ <--- HERE
        loss: torch.Tensor = 1 - ssim_value

```

File: monai/networks/nets/segresnet_ds.py [hunk 2]
```diff
@@ -387,7 +388,7 @@ def is_valid_shape(self, x):
         a = [i % j == 0 for i, j in zip(x.shape[2:], self.shape_factor())]
         return all(a)
 
-    def _forward(self, x: torch.Tensor) -> torch.Tensor | list[torch.Tensor]:
+    def _forward(self, x: torch.Tensor) -> Union[None, torch.Tensor, list[torch.Tensor]]:
```
Comment (yiheng-wang-nv) [inline monai/networks/nets/segresnet_ds.py:391]: Hi @wyli , I tested with `Union[torch.Tensor, list[torch.Tensor]]` and it works. Could I know that what is reason here to add `None`?

Comment (yiheng-wang-nv) [reply monai/networks/nets/segresnet_ds.py:391]: same for `ds_loss`, may also no need to add `None` in type hint. Or will it impact other test functions?
I just run `tests/test_ds_loss.py` and `tests/test_segresnet_ds.py`

Comment (wyli) [reply monai/networks/nets/segresnet_ds.py:391]: The changes are needed for torch<2 and python>3.8, for example https://github.com/Project-MONAI/MONAI/actions/runs/4928134014/jobs/8806104746?pr=6498

Comment (yiheng-wang-nv) [reply monai/networks/nets/segresnet_ds.py:391]: thanks!

**Review by yiheng-wang-nv — APPROVED** [review]

**Pull request merged**